### PR TITLE
Adds initial version of GnuTimeReport

### PR DESCRIPTION
**Pull request by vulder**

### CODE REVIEW

File: varats-core/varats/utils/util.py [hunk 1]
```diff
@@ -0,0 +1,22 @@
...
+
+FunctionType = tp.TypeVar("FunctionType")
+
+
+def static_vars(**kwargs) -> tp.Any:
+    """
+    Decorates a function with static variables, passed with kwargs.
+
+    @static_vars(var_name=value)
+    def func(...):
```
Comment (boehmseb) [inline varats-core/varats/utils/util.py]: Make this an actual code block.

The following should work but is not tested:

```suggestion
    For example::

        @static_vars(var_name=value)
        def func(...):
```

If the highlighting is wrong, an additional `.. code-block:: python` should help.

File: varats-core/varats/report/gnu_time_report.py [hunk 2]
```diff
@@ -0,0 +1,209 @@
...
+    def __init__(self, path: Path) -> None:
+        super().__init__(path)
+
+        with open(self.path, 'r') as stream:
+            for line in stream:
+                line = line.strip()
+
+                if line.startswith("Command being timed"):
+                    self.__command_name: str = ""
+                    self.__command_name = TimeReport._parse_command(line)
```
Comment (boehmseb) [inline varats-core/varats/report/gnu_time_report.py]:
```suggestion
                    self.__command_name = TimeReport._parse_command(line)
```

File: varats-core/varats/report/gnu_time_report.py [hunk 3]
```diff
@@ -0,0 +1,209 @@
...
+
+    @property
+    def max_res_size(self) -> int:
+        """Maximum resident size."""
+        return self.__max_resident_size
+
+    def __str__(self) -> str:
+        return repr(self)
+
+    def __repr__(self) -> str:
```
Comment (boehmseb) [inline varats-core/varats/report/gnu_time_report.py:81]: No time but only resident size in repr?

File: varats-core/varats/report/gnu_time_report.py [hunk 4]
```diff
@@ -0,0 +1,209 @@
...
+        datetime.timedelta(seconds=6121, microseconds=120000)
+
+        >>> import datetime
+        >>> TimeReport._parse_wall_clock_time(\
+                "Elapsed (wall clock) time (h:mm:ss or m:ss): 42:1.12")
+        datetime.timedelta(seconds=2521, microseconds=120000)
+        """
+        match = TimeReport._parse_wall_clock_time.WALL_CLOCK_REGEX.search(line)
+        if match:
+            # t = datetime.strptime(, "%H:%M:%S")
```
Comment (boehmseb) [inline varats-core/varats/report/gnu_time_report.py]:
```suggestion
```

**Review by boehmseb — CHANGES_REQUESTED** [review]

File: varats-core/varats/report/gnu_time_report.py (hunk 3 above)
Comment (vulder) [reply varats-core/varats/report/gnu_time_report.py:81]: I can add everything if you want, just did not see a reason to dump everything.

Comment (boehmseb) [reply varats-core/varats/report/gnu_time_report.py:81]: I don't care if all this information is included in repr or not, but including the resident size but not the other metrics seems inconsistent.

**Review by boehmseb — APPROVED** [review]

**Pull request merged**